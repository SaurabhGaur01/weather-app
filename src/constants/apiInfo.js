export const APP_ID = "75f972b80e26f14fe6c920aa6a85ad57";
export const TEMPERATURE_IN_FAHRENHEIT = "Imperial";
export const TEMPERATURE_IN_CELSIUS = "Metric";
export const COUNT = 40;
export const BASE_URL = "http://api.openweathermap.org/data/2.5/forecast";
export const DEFAULT_LOCATION = "Munich,de";
export const CELSIUS_UNIT = "°C";
export const FAHRENHEIT_UNIT = "°F";