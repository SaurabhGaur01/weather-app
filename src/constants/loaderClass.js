import { css } from "@emotion/core";

export const loaderClass = css`
  display: block;
  margin: 170px auto;
  border-color: red;
`;
export const loaderSize = 60;
export const loaderColor = '#123abc';