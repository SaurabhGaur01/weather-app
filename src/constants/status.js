export const API_CALL_STATUS_SUCCESS = "SUCCESS";
export const API_CALL_STATUS_LOADING = "LOADING";
export const API_CALL_STATUS_ERROR = "ERROR";