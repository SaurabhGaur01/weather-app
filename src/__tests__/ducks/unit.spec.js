import {
    setUnitFahrenheit,
    setUnitCelsius,
    unitReducer
} from '../../ducks/unit';

import {  CELSIUS_UNIT, FAHRENHEIT_UNIT } from '../../constants/apiInfo';

describe('unit Duck Tests', () => {
    const mockState = {
        key: 'mockKey',
    }

    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setUnitCelsius should match the snapshot' , () => {
            expect(setUnitCelsius()).toMatchSnapshot();
        });
        it('setUnitFahrenheit should match the snapshot' , () => {
            expect(setUnitFahrenheit()).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(unitReducer()).toEqual('');
        });
        it('should return the passed state if called with no action' , () => {
            expect(unitReducer(mockState)).toBe(mockState);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(unitReducer(mockState, unknownAction)).toBe(mockState);
        });
    });

    describe('when called with a particular action', () => {
        it('should return the value of the constant FAHRENHEIT_UNIT' , () => {
            expect(unitReducer(mockState, setUnitFahrenheit()))
            .toEqual(FAHRENHEIT_UNIT);
        });
        it('should return the value of the constant CELSIUS_UNIT' , () => {
            expect(unitReducer(mockState, setUnitCelsius()))
            .toEqual(CELSIUS_UNIT);
        });
    });

})