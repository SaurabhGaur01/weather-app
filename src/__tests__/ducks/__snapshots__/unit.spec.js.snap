// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit Duck Tests Action Generator setUnitCelsius should match the snapshot 1`] = `
Object {
  "type": "modules/weather-app/SET_UNIT_CELSIUS",
}
`;

exports[`unit Duck Tests Action Generator setUnitFahrenheit should match the snapshot 1`] = `
Object {
  "type": "modules/weather-app/SET_UNIT_FAHRENHEIT",
}
`;
